app/routes/concept.js

```diff
@@ -16,6 +17,24 @@ export default class ConceptRoute extends BaseRoute {
     });
   }
 
+  async findOrCreateConceptEngagement(concept) {
+    const latestConceptEngagement = this.authenticator.currentUser?.conceptEngagements
+      .filter((engagement) => engagement.concept.slug === concept.slug)
+      .sortBy('createdAt')
+      .reverse()[0];
+
+    if (!latestConceptEngagement) {
+      return await this.store
+        .createRecord('concept-engagement', {
+          concept,
+          user: this.authenticator.currentUser,
+        })
+        .save();
```

Comment:
I don't think we need to `save()` here. I think it'll implicitly get saved when a user ends up clicking on the first block, no? Similar to repositories. We create a repository record when a user starts, but we don't call `save()` until the user chooses their language.

I'm not sure we'd want to count a "concept engagement" if a user only loads the page but does nothing with it.